src/Illuminate/Pipeline/Pipeline.php

```diff
@@ -104,6 +105,21 @@ public function then(Closure $destination)
     }
 
     /**
+     * Get internal container instance.
+     *
+     * @return \Illuminate\Contracts\Container\Container
+     * @throws \RuntimeException
+     */
+    protected function getContainer()
```

Comment:
I think, this may lead to unexpected behavior, e. g. when some class is rebound in the container, but the pipeline was constructed without it. Such things happen quite often in big teams.

We already have `new Class` syntax available, which gives more control over that.